test/e2e/operator/deployment_api.go

```diff
@@ -247,6 +248,24 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 			validateDriver(deployment)
 		})
 
+		It("with JSON logging format", func() {
+			deployment := getDeployment("test-deployment-driver-image")
+			deployment.Spec.Image = ""
+			deployment.Spec.PMEMPercentage = 50
+			deployment.Spec.LogFormat = api.LogFormatJSON
+
+			deployment = deploy.CreateDeploymentCR(f, deployment)
+			defer deploy.DeleteDeploymentCR(f, deployment.Name)
+
+			validateDriver(deployment)
+
+			output, err := e2epod.GetPodLogs(f.ClientSet, "default", deployment.Name+"-controller-0", "pmem-driver")
```

Comment:
Can you please use `d.Namespace` instead of "default", that way this test will work with #829 without and further chagnes.

Reply:
Done.

It also occurred to me that I should be testing *all* log format options (default, text, json), so I added that, too.